src/components/package/edit-custom/custom-package-edit.js

```diff
@@ -82,21 +91,30 @@ class CustomPackageEdit extends Component {
 
   static getDerivedStateFromProps(nextProps, prevState) {
     let stateUpdates = {};
-    const { initialValues } = prevState;
+    const { initialValues, proxyTypesLoaded } = prevState;
     const {
       isSelected,
       destroy,
     } = nextProps.model;
+    const { proxyTypes, provider } = nextProps;
 
     const selectionStatusChanged = isSelected !== initialValues.isSelected;
+    const isProxyTypesLoaded = CustomPackageEdit.isProxyTypesLoaded(proxyTypes, provider);
 
     if (selectionStatusChanged) {
       stateUpdates = {
-        initialValues: CustomPackageEdit.getInitialValues(nextProps.model),
+        initialValues: CustomPackageEdit.getInitialValues(nextProps.model, proxyTypes),
         packageSelected: isSelected
       };
     }
 
+    if (isProxyTypesLoaded && !proxyTypesLoaded) {
```

Comment:
I think it's better to name `proxyTypesLoaded` something like `wasProxyTypesLoaded`, because `isProxyTypesLoaded && !proxyTypesLoaded` is very confusing. Or we could just not destructure `proxyTypesLoaded` and write `isProxyTypesLoaded && !prevState.proxyTypesLoaded`